contrib/IECoreArnold/src/IECoreArnold/RendererImplementation.cpp

```diff
@@ -697,17 +695,12 @@ void IECoreArnold::RendererImplementation::procedural( IECore::Renderer::Procedu
 
 	if( const ExternalProcedural *externalProc = dynamic_cast<ExternalProcedural *>( proc.get() ) )
 	{
-		// Allow a parameter "ai:nodeType" == "volume" to create a volume shape rather
-		// than a procedural shape. Volume shapes provide "dso", "min" and "max" parameters
-		// just as procedural shapes do, so the mapping is a fairly natural one.
-		CompoundDataMap::const_iterator nodeTypeIt = externalProc->parameters().find( "ai:nodeType" );
-		if( nodeTypeIt != externalProc->parameters().end() && nodeTypeIt->second->isInstanceOf( StringData::staticTypeId() ) )
-		{
-			nodeType = static_cast<const StringData *>( nodeTypeIt->second.get() )->readable();
-		}
-		node = AiNode( nodeType.c_str() );
-
-		AiNodeSetStr( node, "dso", externalProc->fileName().c_str() );
+		// \todo In Arnold, external procedurals register node types, and then we use the node types
+		// just like built in nodes - we don't reference the filename of the dso that defines the node type.
+		// So here we just interpret "filename" as the node type to create.
+		// Where should this be documented.  Should we change the name of the parameter to ExternalProcedural
+		// to be something other than "filename" in Cortex 10?
```

Comment:
Seems reasonable, maybe just calling it "name" is generic enough? Let's not mix that up with the Arnold update though - what you have here seems fine for now.